include/glow/Optimizer/GraphOptimizer/PassManager.h

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_OPTIMIZER_GRAPHOPTIMIZER_PASSMANAGER_H
+#define GLOW_OPTIMIZER_GRAPHOPTIMIZER_PASSMANAGER_H
+
+#include "glow/Optimizer/GraphOptimizer/CompilationContext.h"
+#include "glow/Optimizer/GraphOptimizer/FunctionPass.h"
+#include "glow/Optimizer/GraphOptimizer/FunctionPasses.h"
+#include "glow/Optimizer/GraphOptimizer/Pipeline/Pipeline.h"
+
+#include "llvm/ADT/SmallVector.h"
+
+namespace glow {
+
+/// Manager for running a series of FunctionPasses. Given some Function,
+/// CompilationContext, and provided Pipeline, it will run all passes on the
+/// Function. Enables easier debugging given runPrePass() and runPostPass()
+/// calls, which can be modified to run some code before or before every pass.
+class FunctionPassManager : public Named {
```

Comment:
I wonder if we should move this header into a dedicated PassManager directory and prepare it to be a base class for graph-based and IR-based pass managers for optimizations and transformation passes? The reason I suggest to do it now is because it would affects paths and CMake files and thus it is probably easier to do it now, to avoid the updates of all these places in the future.  But this is just a suggestion. I don't have any strong opinions about it.

Reply:
I started to try this, but I'm not sure exactly how it will end up looking once we integrate other passes in. I imagine I'd need to update CMake stuff again either way. So I'd rather just do it on future PRs as the design is fleshed out.